packages/amplify-cli/src/commands/init.ts

```diff
@@ -14,21 +15,33 @@ function constructExeInfo(context: $TSContext) {
   };
 }
 
-export const run = async (context: $TSContext) => {
-  constructExeInfo(context);
-  try {
-    await preInitSetup(context);
-    await analyzeProject(context);
-    await initFrontend(context);
-    await initProviders(context);
-    await onSuccess(context);
-  } catch (e) {
-    context.usageData.emitError(e);
-    onFailure(e);
+const runStrategy = (context: $TSContext) => {
```

Comment:
I would just take in the quickstart bool directly (no need for this function to depend on the whole context object)
```ts
const runStrategy = (quickstart: boolean) => quickstart ? [...quickstart funcs...] : [...other funcs...]
```